### PR TITLE
Automation for CEPH-83594558: Scale-level cloning from group snapshot

### PR DESCRIPTION
**Pull request by aarsharm**
# Description

Automation for CEPH-83594558: Scale-level cloning from group snapshot
Pass log: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-8CD36I

### CODE REVIEW

File: tests/rbd/test_scale_group_snap_clone.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+                    "pool": pool,
+                    "group": group,
+                    "image": image,
+                }
+                rc = add_image_to_group_and_verify(**add_image_group_kw)
+                if rc:
+                    log.error(f"Image {image} add failed, test case fail")
+                    return 1
+                else:
+                    log.info(f"STAGE: image {image} add succeeded")
```
Comment (sunilangadi2) [inline tests/rbd/test_scale_group_snap_clone.py:81]: can it be `STAGE: image {image} added successfully to group`

File: tests/rbd/test_scale_group_snap_clone.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+                "pool": pool,
+                "group": group,
+                "snap": group_snap,
+            }
+            rc = create_snap_and_verify(**snap_create_kw)
+            if rc:
+                log.error(f"snap {group_snap} create failure")
+                return 1
+            else:
+                log.info(f"STAGE: snap {group_snap} creation with validation done")
```
Comment (sunilangadi2) [inline tests/rbd/test_scale_group_snap_clone.py:97]: can it be `STAGE: Group snap {group_snap} created successfully` please avoid generic terms like validation.

File: tests/rbd/test_scale_group_snap_clone.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+            snap_group_info = {
+                "client": client,
+                "pool": pool,
+                "group": group,
+                "snap": group_snap,
+                "format": "json",
+            }
+            (snap_g_out, _) = group_snap_info(**snap_group_info)
+            log.info(snap_g_out)
+            snap_g_out = json.loads(snap_g_out)
```
Comment (sunilangadi2) [inline tests/rbd/test_scale_group_snap_clone.py:129]: can we keep this varibale more meaningful like `snap_group_output`?

File: tests/rbd/test_scale_group_snap_clone.py [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+            # Verify the group snapshot exists using the rbd group snap ls command.
+            log.info(f"Verify group id for group: {group} snap: {group_snap}")
+            group_ls_kw = {
+                "client": client,
+                "pool": pool,
+                "group": group,
+                "format": "json",
+            }
+            (g_ls_out, _) = group_info(**group_ls_kw)
```
Comment (sunilangadi2) [inline tests/rbd/test_scale_group_snap_clone.py:107]: please keep the variable more descriptive here as well `group_output`

File: tests/rbd/test_scale_group_snap_clone.py [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+                out, err = rbd1.info(**info_spec)
+                if err:
+                    log.error(f"Error while fetching info for image {clone}")
+                    return 1
+                out_json = json.loads(out)
+                if out_json["name"] != clone:
+                    log.error(f"Clone image {clone} does not exist")
+                    return 1
+
+            # Map the cloned images as new block disks and run fio using rbd map in a loop.
```
Comment (sunilangadi2) [inline tests/rbd/test_scale_group_snap_clone.py:163]: `run fio using rbd map` sentence looks confusing it should be `run fio on mapped disk`

File: tests/rbd/test_scale_group_snap_clone.py [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+        )
+        ret_val = 1
+
+    finally:
+        cluster_name = kw.get("ceph_cluster", {}).name
+        if "rbd_obj" not in locals():
+            rbd_obj = Rbd(client)
+        obj = {cluster_name: rbd_obj}
+        if pool_types:
+            cleanup(pool_types=pool_types, multi_cluster_obj=obj, **kw)
```
Comment (sunilangadi2) [inline tests/rbd/test_scale_group_snap_clone.py:240]: please check this cleanup removes `ec data pool` as well?

File: suites/squid/rbd/tier-2_rbd_group.yaml [hunk 7]
```diff
@@ -164,3 +164,24 @@ tests:
       module: test_rbd_clone_flatten_group_snapshot.py
       name: Test cloning and flattening group snapshot
       polarion-id: CEPH-83594299
+
+  - test:
```
Comment (sunilangadi2) [inline suites/squid/rbd/tier-2_rbd_group.yaml:168]: we need to keep scale in tier-3 suites can we add this here 
https://github.com/red-hat-storage/cephci/blob/master/suites/squid/rbd/tier-3_rbd_snap_operations.yaml

**Review by sunilangadi2 — CHANGES_REQUESTED** [review]

File: tests/rbd/test_scale_group_snap_clone.py [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+from cli.rbd.rbd import Rbd
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def test_scale_group_snap_clone(rbd_obj, client, **kw):
+    """
+    Tests the group creation followed by cloning image from group snapshot at scale
+    Args:
```
[the remark this answers is not in the document]
Comment (aarsharm) [reply tests/rbd/test_scale_group_snap_clone.py:25]: Done

File: tests/rbd/test_scale_group_snap_clone.py (hunk 1 above)
Comment (aarsharm) [reply tests/rbd/test_scale_group_snap_clone.py:81]: Done

File: tests/rbd/test_scale_group_snap_clone.py (hunk 2 above)
Comment (aarsharm) [reply tests/rbd/test_scale_group_snap_clone.py:97]: Done

File: tests/rbd/test_scale_group_snap_clone.py (hunk 3 above)
Comment (aarsharm) [reply tests/rbd/test_scale_group_snap_clone.py:129]: Done

File: tests/rbd/test_scale_group_snap_clone.py (hunk 4 above)
Comment (aarsharm) [reply tests/rbd/test_scale_group_snap_clone.py:107]: Done

File: tests/rbd/test_scale_group_snap_clone.py (hunk 5 above)
Comment (aarsharm) [reply tests/rbd/test_scale_group_snap_clone.py:163]: Done

File: tests/rbd/test_scale_group_snap_clone.py (hunk 6 above)
Comment (aarsharm) [reply tests/rbd/test_scale_group_snap_clone.py:240]: Done

File: suites/squid/rbd/tier-2_rbd_group.yaml (hunk 7 above)
Comment (aarsharm) [reply suites/squid/rbd/tier-2_rbd_group.yaml:168]: Sure, Done

**Comment by aarsharm** [comment]
Pass logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-PSRRJY

**Comment by sunilangadi2** [comment]
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-PSRRJY/RBD_group_image_clone_create_0.log
EC related data pool not getting deleted here please check that.

**Comment by aarsharm** [comment]
Pass log: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-7EMZZP

**Review by sunilangadi2 — APPROVED** [review]
LGTM.

**Review by sunilkumarn417 — APPROVED** [review]

**Pull request merged**